src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,10 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value' && node.hasAttribute('value')) {
```

Comment:
Type coercion solved my problems, however it makes the initial case hard. Empty inputs are always equal to `0`. 

I think this could be better. Any suggestions?